internal/storage/fs/git/source.go

```diff
@@ -64,6 +67,30 @@ func WithAuth(auth transport.AuthMethod) containers.Option[Source] {
 	}
 }
 
+// WithInsecureTLS returns an option which configures the insecure TLS
+// setting for the provided source.
+func WithInsecureTLS(insecureSkipTLS bool) containers.Option[Source] {
+	return func(s *Source) {
+		s.insecureSkipTLS = insecureSkipTLS
+	}
+}
+
+// WithCABundle returns an option which configures the CA Bundle used for
+// validating the TLS connection to the provided source.
+func WithCABundle(caCertBytes []byte) containers.Option[Source] {
+	return func(s *Source) {
+		if caCertBytes != nil {
+			s.caBundle = caCertBytes
+		}
+	}
+}
+
+// CaBundleFromFile tries to load an x.509 CA certificate from a path
+func CaBundleFromFile(caPath string) ([]byte, error) {
```

Comment:
doesnt look like theres any thing `CA` specific for this func as its really just wrapping `os.ReadFile`, we could probably just remove this and call `os.ReadFile` directly in `grpc.go`:

https://github.com/flipt-io/flipt/pull/2443/files#diff-b3edacef8342d287b8d908a586f22acedfa5f180d94b7a84e665e7f7a2cc973cR166